packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -1,66 +1,191 @@
+  id: string
+  name: string
+  addresses: Addresses
+
+  private storePath: string
+
+  constructor(id: string, props: WalletProperties, storePath: string) {
+    this.id = id
+
+    this.name = props.name
+    this.addresses = props.addresses
+
+    this.storePath = storePath
+  }
+
+  static fromJSON = (
+    json: { id: string; name: string; addresses: Addresses },
+    storePath: string,
+  ): FileKeystoreWallet => {
+    const props = { name: json.name, addresses: json.addresses, keystore: null }
+    return new FileKeystoreWallet(json.id, props, storePath)
+  }
+
+  update = (props: WalletProperties) => {
+    this.name = props.name
+    this.addresses = props.addresses
+  }
 
-const walletStore = new WalletStore()
+  toJSON = (): any => {
+    return {
+      id: this.id,
+      name: this.name,
+      addresses: this.addresses,
+    }
+  }
+
+  loadKeystore = (): Keystore => {
+    // TODO: handle fs error
+    const data = fs.readFileSync(this.storeLocation(), { encoding: 'utf8' })
+    return JSON.parse(data) as Keystore
+  }
+
+  private storeLocation = (): string => {
+    return path.resolve(this.storePath, `${this.id}.json`)
+  }
+
+  saveKeystore = (keystore: Keystore) => {
+    fs.writeFileSync(this.storeLocation(), JSON.stringify(keystore), { encoding: 'utf8' })
+  }
+
+  deleteKeystore = () => {
+    fs.unlinkSync(this.storeLocation())
+  }
+}
+
+// TODO: Check if '/dev/wallets' path works on Windows
+const defaultStorePath = env.isDevMode ? '/dev/wallets' : '/wallets'
 
 export default class WalletService {
-  public getAll = (): WalletData[] => {
-    return walletStore.getAllWallets()
+  private storePath: string
+  private listStore: Store // Save wallets (meta info except keystore, which is persisted separately)
+  private walletsKey = 'wallets'
+  private currentWalletKey = 'current'
+
+  constructor(storePath: string = defaultStorePath) {
+    this.storePath = `${app.getPath('userData')}/${storePath}`
+    fs.mkdirSync(this.storePath, { recursive: true })
+    this.listStore = new Store(this.storePath, 'wallets.json')
   }
 
-  public get = (id: string): WalletData | undefined => {
-    return this.getAll().find(wallet => wallet.id === id)
+  public getAll = (): Wallet[] => {
+    return this.listStore.readSync(this.walletsKey) || []
   }
 
-  public create = ({
-    name,
-    keystore,
-    addresses,
-  }: {
-    name: string
-    keystore: Keystore
-    addresses: Addresses
-  }): WalletData => {
-    const index = walletStore.getAllWallets().findIndex(wallet => wallet.name === name)
-    if (index !== -1) {
-      throw Error('Wallet name existed')
+  public get = (id: string): Wallet | undefined => {
+    const wallet = this.getAll().find(w => w.id === id)
+    if (wallet) {
+      return FileKeystoreWallet.fromJSON(wallet, this.storePath)
     }
-    const id = v4()
-    walletStore.saveWallet({ id, name, keystore, addresses })
-    return { id, name, keystore, addresses }
+    return undefined
```

Comment:
better to use `null` which explicitly means `none`

Reply:
Makes sense. Doing that in a future refactoring.